// cloudflare/resources/zones/settings/settings.mjs
// File generated from our OpenAPI spec by Stainless. See CONTRIBUTING.md for details.
import { APIResource } from "../../../resource.mjs";
import * as AdvancedDDoSAPI from "./advanced-ddos.mjs";
import * as AlwaysOnlineAPI from "./always-online.mjs";
import * as AlwaysUseHTTPSAPI from "./always-use-https.mjs";
import * as AutomaticHTTPSRewritesAPI from "./automatic-https-rewrites.mjs";
import * as AutomaticPlatformOptimizationAPI from "./automatic-platform-optimization.mjs";
import * as BrotliAPI from "./brotli.mjs";
import * as BrowserCacheTTLAPI from "./browser-cache-ttl.mjs";
import * as BrowserCheckAPI from "./browser-check.mjs";
import * as CacheLevelAPI from "./cache-level.mjs";
import * as ChallengeTTLAPI from "./challenge-ttl.mjs";
import * as CiphersAPI from "./ciphers.mjs";
import * as DevelopmentModeAPI from "./development-mode.mjs";
import * as EarlyHintsAPI from "./early-hints.mjs";
import * as EmailObfuscationAPI from "./email-obfuscation.mjs";
import * as FontSettingsAPI from "./font-settings.mjs";
import * as H2PrioritizationAPI from "./h2-prioritization.mjs";
import * as HotlinkProtectionAPI from "./hotlink-protection.mjs";
import * as HTTP2API from "./http2.mjs";
import * as HTTP3API from "./http3.mjs";
import * as ImageResizingAPI from "./image-resizing.mjs";
import * as IPGeolocationAPI from "./ip-geolocation.mjs";
import * as IPV6API from "./ipv6.mjs";
import * as MinTLSVersionAPI from "./min-tls-version.mjs";
import * as MinifyAPI from "./minify.mjs";
import * as MirageAPI from "./mirage.mjs";
import * as MobileRedirectAPI from "./mobile-redirect.mjs";
import * as NELAPI from "./nel.mjs";
import * as OpportunisticEncryptionAPI from "./opportunistic-encryption.mjs";
import * as OpportunisticOnionAPI from "./opportunistic-onion.mjs";
import * as OrangeToOrangeAPI from "./orange-to-orange.mjs";
import * as OriginErrorPagePassThruAPI from "./origin-error-page-pass-thru.mjs";
import * as OriginMaxHTTPVersionAPI from "./origin-max-http-version.mjs";
import * as PolishAPI from "./polish.mjs";
import * as PrefetchPreloadAPI from "./prefetch-preload.mjs";
import * as ProxyReadTimeoutAPI from "./proxy-read-timeout.mjs";
import * as PseudoIPV4API from "./pseudo-ipv4.mjs";
import * as ResponseBufferingAPI from "./response-buffering.mjs";
import * as RocketLoaderAPI from "./rocket-loader.mjs";
import * as SecurityHeadersAPI from "./security-headers.mjs";
import * as SecurityLevelAPI from "./security-level.mjs";
import * as ServerSideExcludesAPI from "./server-side-excludes.mjs";
import * as SortQueryStringForCacheAPI from "./sort-query-string-for-cache.mjs";
import * as SSLAPI from "./ssl.mjs";
import * as SSLRecommenderAPI from "./ssl-recommender.mjs";
import * as TLS1_3API from "./tls-1-3.mjs";
import * as TLSClientAuthAPI from "./tls-client-auth.mjs";
import * as TrueClientIPHeaderAPI from "./true-client-ip-header.mjs";
import * as WAFAPI from "./waf.mjs";
import * as WebPAPI from "./webp.mjs";
import * as WebsocketAPI from "./websocket.mjs";
import * as ZeroRTTAPI from "./zero-rtt.mjs";
export class Settings extends APIResource {
    constructor() {
        super(...arguments);
        this.zeroRTT = new ZeroRTTAPI.ZeroRTTResource(this._client);
        this.advancedDDoS = new AdvancedDDoSAPI.AdvancedDDoS(this._client);
        this.alwaysOnline = new AlwaysOnlineAPI.AlwaysOnlineResource(this._client);
        this.alwaysUseHTTPS = new AlwaysUseHTTPSAPI.AlwaysUseHTTPS(this._client);
        this.automaticHTTPSRewrites = new AutomaticHTTPSRewritesAPI.AutomaticHTTPSRewrites(this._client);
        this.automaticPlatformOptimization = new AutomaticPlatformOptimizationAPI.AutomaticPlatformOptimizationResource(this._client);
        this.brotli = new BrotliAPI.BrotliResource(this._client);
        this.browserCacheTTL = new BrowserCacheTTLAPI.BrowserCacheTTLResource(this._client);
        this.browserCheck = new BrowserCheckAPI.BrowserCheckResource(this._client);
        this.cacheLevel = new CacheLevelAPI.CacheLevelResource(this._client);
        this.challengeTTL = new ChallengeTTLAPI.ChallengeTTLResource(this._client);
        this.ciphers = new CiphersAPI.Ciphers(this._client);
        this.developmentMode = new DevelopmentModeAPI.DevelopmentModeResource(this._client);
        this.earlyHints = new EarlyHintsAPI.EarlyHints(this._client);
        this.emailObfuscation = new EmailObfuscationAPI.EmailObfuscationResource(this._client);
        this.h2Prioritization = new H2PrioritizationAPI.H2PrioritizationResource(this._client);
        this.hotlinkProtection = new HotlinkProtectionAPI.HotlinkProtectionResource(this._client);
        this.http2 = new HTTP2API.HTTP2Resource(this._client);
        this.http3 = new HTTP3API.HTTP3Resource(this._client);
        this.imageResizing = new ImageResizingAPI.ImageResizingResource(this._client);
        this.ipGeolocation = new IPGeolocationAPI.IPGeolocationResource(this._client);
        this.ipv6 = new IPV6API.IPV6Resource(this._client);
        this.minTLSVersion = new MinTLSVersionAPI.MinTLSVersionResource(this._client);
        this.minify = new MinifyAPI.MinifyResource(this._client);
        this.mirage = new MirageAPI.MirageResource(this._client);
        this.mobileRedirect = new MobileRedirectAPI.MobileRedirectResource(this._client);
        this.nel = new NELAPI.NELResource(this._client);
        this.opportunisticEncryption = new OpportunisticEncryptionAPI.OpportunisticEncryptionResource(this._client);
        this.opportunisticOnion = new OpportunisticOnionAPI.OpportunisticOnionResource(this._client);
        this.orangeToOrange = new OrangeToOrangeAPI.OrangeToOrangeResource(this._client);
        this.originErrorPagePassThru = new OriginErrorPagePassThruAPI.OriginErrorPagePassThruResource(this._client);
        this.originMaxHTTPVersion = new OriginMaxHTTPVersionAPI.OriginMaxHTTPVersionResource(this._client);
        this.polish = new PolishAPI.PolishResource(this._client);
        this.prefetchPreload = new PrefetchPreloadAPI.PrefetchPreloadResource(this._client);
        this.proxyReadTimeout = new ProxyReadTimeoutAPI.ProxyReadTimeoutResource(this._client);
        this.pseudoIPV4 = new PseudoIPV4API.PseudoIPV4Resource(this._client);
        this.responseBuffering = new ResponseBufferingAPI.ResponseBufferingResource(this._client);
        this.rocketLoader = new RocketLoaderAPI.RocketLoaderResource(this._client);
        this.securityHeaders = new SecurityHeadersAPI.SecurityHeaders(this._client);
        this.securityLevel = new SecurityLevelAPI.SecurityLevelResource(this._client);
        this.serverSideExcludes = new ServerSideExcludesAPI.ServerSideExcludes(this._client);
        this.sortQueryStringForCache = new SortQueryStringForCacheAPI.SortQueryStringForCacheResource(this._client);
        this.ssl = new SSLAPI.SSLResource(this._client);
        this.sslRecommender = new SSLRecommenderAPI.SSLRecommenderResource(this._client);
        this.tls1_3 = new TLS1_3API.TLS1_3Resource(this._client);
        this.tlsClientAuth = new TLSClientAuthAPI.TLSClientAuthResource(this._client);
        this.trueClientIPHeader = new TrueClientIPHeaderAPI.TrueClientIPHeaderResource(this._client);
        this.waf = new WAFAPI.WAFResource(this._client);
        this.WebP = new WebPAPI.WebPResource(this._client);
        this.websocket = new WebsocketAPI.WebsocketResource(this._client);
        this.fontSettings = new FontSettingsAPI.FontSettings(this._client);
    }
}
(function (Settings) {
    Settings.ZeroRTTResource = ZeroRTTAPI.ZeroRTTResource;
    Settings.AdvancedDDoS = AdvancedDDoSAPI.AdvancedDDoS;
    Settings.AlwaysOnlineResource = AlwaysOnlineAPI.AlwaysOnlineResource;
    Settings.AlwaysUseHTTPS = AlwaysUseHTTPSAPI.AlwaysUseHTTPS;
    Settings.AutomaticHTTPSRewrites = AutomaticHTTPSRewritesAPI.AutomaticHTTPSRewrites;
    Settings.AutomaticPlatformOptimizationResource = AutomaticPlatformOptimizationAPI.AutomaticPlatformOptimizationResource;
    Settings.BrotliResource = BrotliAPI.BrotliResource;
    Settings.BrowserCacheTTLResource = BrowserCacheTTLAPI.BrowserCacheTTLResource;
    Settings.BrowserCheckResource = BrowserCheckAPI.BrowserCheckResource;
    Settings.CacheLevelResource = CacheLevelAPI.CacheLevelResource;
    Settings.ChallengeTTLResource = ChallengeTTLAPI.ChallengeTTLResource;
    Settings.Ciphers = CiphersAPI.Ciphers;
    Settings.DevelopmentModeResource = DevelopmentModeAPI.DevelopmentModeResource;
    Settings.EarlyHints = EarlyHintsAPI.EarlyHints;
    Settings.EmailObfuscationResource = EmailObfuscationAPI.EmailObfuscationResource;
    Settings.H2PrioritizationResource = H2PrioritizationAPI.H2PrioritizationResource;
    Settings.HotlinkProtectionResource = HotlinkProtectionAPI.HotlinkProtectionResource;
    Settings.HTTP2Resource = HTTP2API.HTTP2Resource;
    Settings.HTTP3Resource = HTTP3API.HTTP3Resource;
    Settings.ImageResizingResource = ImageResizingAPI.ImageResizingResource;
    Settings.IPGeolocationResource = IPGeolocationAPI.IPGeolocationResource;
    Settings.IPV6Resource = IPV6API.IPV6Resource;
    Settings.MinTLSVersionResource = MinTLSVersionAPI.MinTLSVersionResource;
    Settings.MinifyResource = MinifyAPI.MinifyResource;
    Settings.MirageResource = MirageAPI.MirageResource;
    Settings.MobileRedirectResource = MobileRedirectAPI.MobileRedirectResource;
    Settings.NELResource = NELAPI.NELResource;
    Settings.OpportunisticEncryptionResource = OpportunisticEncryptionAPI.OpportunisticEncryptionResource;
    Settings.OpportunisticOnionResource = OpportunisticOnionAPI.OpportunisticOnionResource;
    Settings.OrangeToOrangeResource = OrangeToOrangeAPI.OrangeToOrangeResource;
    Settings.OriginErrorPagePassThruResource = OriginErrorPagePassThruAPI.OriginErrorPagePassThruResource;
    Settings.OriginMaxHTTPVersionResource = OriginMaxHTTPVersionAPI.OriginMaxHTTPVersionResource;
    Settings.PolishResource = PolishAPI.PolishResource;
    Settings.PrefetchPreloadResource = PrefetchPreloadAPI.PrefetchPreloadResource;
    Settings.ProxyReadTimeoutResource = ProxyReadTimeoutAPI.ProxyReadTimeoutResource;
    Settings.PseudoIPV4Resource = PseudoIPV4API.PseudoIPV4Resource;
    Settings.ResponseBufferingResource = ResponseBufferingAPI.ResponseBufferingResource;
    Settings.RocketLoaderResource = RocketLoaderAPI.RocketLoaderResource;
    Settings.SecurityHeaders = SecurityHeadersAPI.SecurityHeaders;
    Settings.SecurityLevelResource = SecurityLevelAPI.SecurityLevelResource;
    Settings.ServerSideExcludes = ServerSideExcludesAPI.ServerSideExcludes;
    Settings.SortQueryStringForCacheResource = SortQueryStringForCacheAPI.SortQueryStringForCacheResource;
    Settings.SSLResource = SSLAPI.SSLResource;
    Settings.SSLRecommenderResource = SSLRecommenderAPI.SSLRecommenderResource;
    Settings.TLS1_3Resource = TLS1_3API.TLS1_3Resource;
    Settings.TLSClientAuthResource = TLSClientAuthAPI.TLSClientAuthResource;
    Settings.TrueClientIPHeaderResource = TrueClientIPHeaderAPI.TrueClientIPHeaderResource;
    Settings.WAFResource = WAFAPI.WAFResource;
    Settings.WebPResource = WebPAPI.WebPResource;
    Settings.WebsocketResource = WebsocketAPI.WebsocketResource;
    Settings.FontSettings = FontSettingsAPI.FontSettings;
})(Settings || (Settings = {}));
